driver-core/src/main/java/com/datastax/driver/core/policies/ErrorAwarePolicy.java

```diff
@@ -0,0 +1,313 @@
+        }
+
+        /**
+         * Creates the {@link ErrorAwarePolicy} instance.
+         * @return the newly created {@link ErrorAwarePolicy}.
+         */
+        public ErrorAwarePolicy build() {
+            return new ErrorAwarePolicy(this);
+        }
+    }
+
+    class PerHostErrorTracker implements LatencyTracker {
+
+        private double errorsThreshold;
+
+        private ConcurrentMap<Host, Meter> hostsMeters = new ConcurrentHashMap<Host, Meter>();
+
+        private ConcurrentMap<Host, Long> exclusions = new ConcurrentHashMap<Host, Long>();
+
+        private ErrorsFilter errorsFilter;
+
+        PerHostErrorTracker(double errorsThreshold, ErrorsFilter errorsFilter) {
+            this.errorsThreshold = errorsThreshold;
+            this.errorsFilter = errorsFilter;
+        }
+
+        @Override
+        public void update(Host host, Statement statement, Exception exception, long newLatencyNanos) {
+            if (exception == null) {
+                return;
+            }
+            if (!errorsFilter.shouldAccountForError(exception, host)) {
+                return;
+            }
+            if (!hostsMeters.containsKey(host)) {
+                hostsMeters.putIfAbsent(host, new Meter());
+            }
+            Meter hostMeter = hostsMeters.get(host);
+            hostMeter.mark();
+            if (hostMeter.getOneMinuteRate() >= errorsThreshold) {
+                logger.debug(String.format("Host : %s has reached the maximum authorized number of errors. Its error rate is : %.3f. " +
+                                "It will now be ignored by the ErrorAware load balancing policy during a period of %d milliseconds.",
+                        host, hostMeter.getOneMinuteRate(), retryPeriodMillis));
+                exclusions.putIfAbsent(host, System.currentTimeMillis());
+            }
+        }
+
+        @Override
+        public void onRegister(Cluster cluster) {
+            // hurray !
+        }
+
+        @Override
+        public void onUnregister(Cluster cluster) {
+            // ...bummer
+        }
+
+        Map<Host, Long> getExclusions() {
+            return this.exclusions;
+        }
+
+        Map<Host, Meter> getHostsMeters() {
+            return this.hostsMeters;
+        }
+    }
+
+    static class DefaultErrorsFilter implements ErrorsFilter {
+        @Override
+        public boolean shouldAccountForError(Exception e, Host host) {
+            return !UnavailableException.class.isAssignableFrom(e.getClass())
+                    && !ReadTimeoutException.class.isAssignableFrom(e.getClass())
+                    && !WriteTimeoutException.class.isAssignableFrom(e.getClass())
+                    && !ReadFailureException.class.isAssignableFrom(e.getClass())
+                    && !WriteFailureException.class.isAssignableFrom(e.getClass());
+        }
+    }
+
+    public interface ErrorsFilter {
+        public boolean shouldAccountForError(Exception e, Host host);
```

Comment:
`shouldAccountForError` seems appropriate to me. `shouldConsiderError` or `shouldIncludeError` do seem like they could be more clear though.